packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -751,15 +752,33 @@ export class ContextualMenuBase extends BaseComponent<IContextualMenuProps, ICon
   };
 
   private _shouldHandleKeyUp = (ev: React.KeyboardEvent<HTMLElement>) => {
-    const shouldHandleKey = this._processingExpandCollapseKeyOnly && this._isExpandCollapseKey(ev);
-    this._processingExpandCollapseKeyOnly = false;
-    return shouldHandleKey;
+    // We close the menu on key up only if the most recent key *down* was alt or meta (command) and
+    // it was *not* followed by some other key (such as up/down arrow to collapse/expand the menu),
+    // AND we're not on a Mac. This is because on Windows, pressing alt moves focus to the application
+    // menu bar or similar, closing any open context menus. There is not a similar behavior on Macs.
+    const keyPressIsAltOrMetaAlone = this._lastKeyDownWasAltOrMeta && this._isAltOrMeta(ev);
+    this._lastKeyDownWasAltOrMeta = false;
+
+    // This will catch Mac desktop as well as iPhone/iPad with external keyboard
+    // (iPhone/iPad have "like Mac OS X" in user agent).
+    const isMac = typeof navigator !== 'undefined' && navigator.userAgent.indexOf('Mac OS X') >= 0;
+
```

Comment:
This only needs to be run once, correct?

Also, shouldn't it be `indexOf('Macintosh')`? Currently this is only going to be true for OS X

Reply:
I'm now handling this differently, but I'm not sure what other versions of Mac OS we'd be concerned about?